mlflow/R/mlflow/R/tracking-client.R

```diff
@@ -0,0 +1,371 @@
+new_mlflow_client <- function(tracking_uri, server_url = NULL) {
+  structure(
+    list(
+      tracking_uri = tracking_uri,
+      server_url = server_url %||% tracking_uri
+    ),
+    class = "mlflow_client"
+  )
+}
+
+#' Initialize an MLflow client
+#'
+#' @param tracking_uri The tracking URI
+#'
+#' @export
+mlflow_client <- function(tracking_uri = NULL) {
+  tracking_uri <- tracking_uri %||% mlflow_get_tracking_uri()
+  server_url <- if (startsWith(tracking_uri, "http")) {
+    tracking_uri
+  } else if (!is.null(mlflow_local_server(tracking_uri)$server_url)) {
+    mlflow_local_server(tracking_uri)$server_url
+  } else {
+    local_server <- mlflow_server(file_store = tracking_uri, port = mlflow_connect_port())
+    mlflow_register_local_server(tracking_uri = tracking_uri, local_server = local_server)
+    local_server$server_url
+  }
+
+  new_mlflow_client(tracking_uri, server_url = server_url)
+}
+
+#' Create Experiment
+#'
+#' Creates an MLflow experiment.
+#'
+#' @param name The name of the experiment to create.
+#' @param artifact_location Location where all artifacts for this experiment are stored. If
+#'   not provided, the remote server will select an appropriate default.
+#' @template roxlate-client-optional
+#'
+#' @export
+mlflow_create_experiment <- function(name, artifact_location = NULL, client = NULL) {
+  UseMethod("mlflow_create_experiment", client)
+}
+
+#' @export
+mlflow_create_experiment.mlflow_client <- function(name, artifact_location = NULL, client = NULL) {
+  name <- forge::cast_string(name)
+  response <- mlflow_client_create_experiment(client, name, artifact_location)
+  invisible(response$experiment_id)
+}
+
+#' Create Run
+#'
+#' reate a new run within an experiment. A run is usually a single execution of a machine learning or data ETL pipeline.
+#'
+#' MLflow uses runs to track Param, Metric, and RunTag, associated with a single execution.
+#'
+#' @param experiment_id Unique identifier for the associated experiment.
+#' @param user_id User ID or LDAP for the user executing the run.
+#' @param run_name Human readable name for run.
+#' @param source_type Originating source for this run. One of Notebook, Job, Project, Local or Unknown.
+#' @param source_name String descriptor for source. For example, name or description of the notebook, or job name.
+#' @param start_time Unix timestamp of when the run started in milliseconds.
+#' @param source_version Git version of the source code used to create run.
+#' @param entry_point_name Name of the entry point for the run.
+#' @param tags Additional metadata for run in key-value pairs.
+#' @template roxlate-client
+#' @export
+mlflow_create_run <- function(
```

Comment:
Oh actually I think I understand after reading the fluent section. 

Hmm I am not sure if method overloading on client==NULL (if I understand correctly) is a good idea in this case, because one version has a run_id as a mandatory argument while the other ignores it and uses the active run. Maybe I don't understand correctly, but what about the following code:
`
`
`
mlflow_start_run() # active run == run1
`
`
mlflow_log_metric(key="metric1", value=1, run_id="run2", client = mlflow_client()) # this should be logged under run2 I believe?`
`
mlflow_log_metric(key="metric1", value=1, run_id="run2") # is this logged under run1?
`
`
`
If I understand it correctly, both of the lines above would work but they would log the metric under different run?  I think it might be better to explicitly call mlflow_client_log_metric() if that is the case.